### PR TITLE
`gfx.h`

### PR DESCRIPTION
**Pull request by AngheloAlf**
- Moves all the "gfx" structs from `z64.h` to a dedicated header.
- Moves the gfx related macros from `macros.h` to this header
- Move the functions and variables from `z_rcp.c` to this header. I'm not completely sure about this change, we may want to revert it.
- On unrelated stuff: I noticed `IRQ_MSG_TYPE` and `IRQ_TYPE` are unused, so I removed them

### CODE REVIEW

**Review by mzxrules — APPROVED** [review]
Looks good enough for me for now, though I feel like we should have a convo on discord about GfxMasterList stuff to come up with some better names.

**Comment by AngheloAlf** [comment]
> Looks good enough for me for now, though I feel like we should have a convo on discord about GfxMasterList stuff to come up with some better names.

Yeah I agree, a lot of the gfx stuff needs to be renamed, but a lot of it goes over my head

File: include/gfx.h [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    /* 0x058 */ OSMesgQueue* schedMsgQ;
+    /* 0x05C */ OSMesgQueue queue;
+    /* 0x074 */ char        unk_74[0x04];
+    /* 0x078 */ OSScTask    task;
+    /* 0x0E0 */ char        unk_E0[0xD0];
+    /* 0x1B0 */ Gfx*        workBuffer;
+    /* 0x1B4 */ TwoHeadGfxArena work;
+    /* 0x1C4 */ Gfx*        debugBuffer;
+    /* 0x1C8 */ TwoHeadGfxArena debug;
+    /* 0x1D8 */ char        unk_1D8[0xAC];
```
Comment (engineer124) [inline include/gfx.h:58]:
```suggestion
    /* 0x1D8 */ UNK_TYPE1 unk_1D8[0xAC];
```
These seem unused? Applies to a few. Could always do this cleanup later though

File: include/gfx.h [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+#include "alignment.h"
+
+
+struct GameState;
+
+typedef struct {
+    /* 0x000 */ Gfx taskStart[9];
+    /* 0x048 */ Gfx clearZBuffer[8];       // original name: clear_zb_dl
+    /* 0x088 */ Gfx clearFrameBuffer[5];   // original name: clear_fb_dl
+    /* 0x0B0 */ Gfx setupBuffers[6];       // setup framebuffer and zbuffer
```
Comment (engineer124) [inline include/gfx.h:16]: Spacing of comments doesn't need to be so large

**Review by engineer124 — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Pull request merged**